src/keys/msgsig.rs

```diff
@@ -0,0 +1,128 @@
+
+/// Signing arbitrary messages and verify message signatures
+
+use bitcoin::hashes::Hash;
+use bitcoin::secp256k1::recovery::RecoverableSignature;
+use bitcoin::secp256k1::{All, Message, Secp256k1, SecretKey};
+use bitcoin::util::misc::{signed_msg_hash, MessageSignature};
+use bitcoin::{Address, Network, PrivateKey, PublicKey};
+
+/// Trait for message signers
+pub trait MessageSigner<S> {
```

Comment:
Are there any other signature types you'd plan to support on top of `RecoverableSignature`? I don't really understand why you need the generic here and on the verifier

Reply:
Yes, exactly. The idea behind this is to add other signature types.

Reply:
As far as I am aware there are no plans to switch to schnorr signatures for Bitcoin signed messages, so I don't think there's really too much room for upgrades here.

I was asking if you had at least one specific use case in memory (like a BIP which uses other sig algorithms) but if you don't have any examples I would just say it's better to drop the generic and stick to the classic ecdsa algorithm.

Reply:
No, I had no other use cases in mind than multi-sig and possibly schonrr. But since the latter is not planned and multi-sig can be done with a map as suggested by you, I agree it's better to just add a sign method to the wallet.